modules/affinity/src/main/java/org/apache/ignite/internal/affinity/Assignments.java

```diff
@@ -53,32 +53,38 @@ public class Assignments implements Serializable {
      */
     private final boolean force;
 
+    /**
+     * Version of Catalog that matches current assignments.
+     */
+    private final int catalogVersion;
```

Comment:
I don't like how it looks. Formally Assignments is a part of affinity module that should not depend on catalog. Let's revise it one more time. Let's revise an idea of using a timestamp here.